unit-tests/src/main/scala/java/util/regex/PatternSuite.scala

```diff
@@ -0,0 +1,259 @@
+package java.util
+package regex
+
+import scala.collection.immutable.List
+
+object PatternSuite extends tests.Suite {
+  test("split/split(n)") {
+    val input = "boo:and:foo"
+
+    val p1 = Pattern.compile(":")
+    assertEquals(p1.split(input).toList, List("boo", "and", "foo"))
+    assertEquals(p1.split(input, 2).toList, List("boo", "and:foo"))
+    assertEquals(p1.split(input, 5).toList, List("boo", "and", "foo"))
+    assertEquals(p1.split(input, -2).toList, List("boo", "and", "foo"))
+
+    val p2 = Pattern.compile("o")
+    assertEquals(p2.split(input).toList, List("b", "", ":and:f"))
+    assertEquals(p2.split(input, 5).toList, List("b", "", ":and:f", "", ""))
+    assertEquals(p2.split(input, -2).toList, List("b", "", ":and:f", "", ""))
+    assertEquals(p2.split(input, 0).toList, List("b", "", ":and:f"))
+  }
+
+  test("quote") {
+    assertEquals(
+      Pattern.quote("1.5-2.0?"),
+      "1\\.5\\-2\\.0\\?"
+    )
+  }
+
+  /*
+  test("characters") {
```

Comment:
We generally don't accept PRs with commented out code unless it's a signature for the method we haven't implemented yet.